### PR TITLE
Don't store/restore private downloads

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt [hunk 1]
```diff
@@ -103,16 +104,17 @@ class DownloadMiddleware(
     private fun restoreDownloads(store: Store<BrowserState, BrowserAction>) = scope.launch {
         downloadStorage.getDownloads().collect { downloads ->
             downloads.forEach { download ->
-                if (!store.state.downloads.containsKey(download.id)) {
+                if (!store.state.downloads.containsKey(download.id) && !download.private) {
```
Comment (pocmo) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt:107]: If we never save them and never restore them... then we do not need the private flag in the database anymore? Should we drop it and as part of the database migration also drop all downloads with private = true from the database?

**Review by pocmo — APPROVED** [review]

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt (hunk 1 above)
Comment (Amejia481) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt:107]: 👍

**Comment by Amejia481** [comment]
bors try

**Comment by Amejia481** [comment]
bors try

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt (hunk 1 above)
Comment (Amejia481) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt:107]: I pushed it here https://github.com/mozilla-mobile/android-components/pull/8526/commits/edcb645aeb03ee57f8f7b6b714e9f216d0548f0e :)

**Comment by Amejia481** [comment]
image

-  For the `build-feature-recentlyclosed` intermittent-test I opened  https://github.com/mozilla-mobile/android-components/issues/8530

- For `test-android-feature-top-sites ` Task timeout after 2400 seconds

-  For `test-ui-browser` I was able to run the test locally.

**Comment by Amejia481** [comment]
bors try

**Comment by Amejia481** [comment]
As the failures are not related to this change I think is safe to land

**Pull request merged**